onmt/test_function/test_inference.py

```diff
@@ -0,0 +1,56 @@
+from onmt.inference import predict_file
+from pathlib import Path
+
+img_fn = "./images/image_1.png"
+
+def test_infer(n_image, batch_size, type_input=str):
+    file_names = [type_input(img_fn)] * n_image
+
+    return predict_file(file_name=file_names, batch_size=batch_size)
+
+"""
+In case the file names collection contains broken file name
+"""
+def test_with_broken_fn(n_image, batch_size, broken_fn):
+    file_names = [img_fn] * n_image
+    file_names += [broken_fn]
+
+    return predict_file(file_name=file_names, batch_size=batch_size)
+
+if __name__ == "__main__":
+    import time
+
+    # 1 image, batch size 1, input type is string
+    print ("\n-- test inference with 1 image, batch size 1, input type is string")
+    result = test_infer(n_image=1, batch_size=1)
+    print (result)
+
+    # 1 image, batch size 1, input type is Path
+    print("\n-- test inference with 1 image, batch size 1, input type is Path")
+    result = test_infer(n_image=1, batch_size=1, type_input=Path)
+    print(result)
+
+    # 1 image, batch size > 1
+    print("\n-- test inference with 1 image, batch size 2.")
+    result = test_infer(n_image=1, batch_size=2)
+    print (result)
+
```

Comment:
We need a case for single element list input
```suggestion

    # list of 1 image, batch size > 1
    print("\n-- test inference with list of 1 image, batch size 1.")
    result = predict_file([img_fn], 1)
    assert isinstance(result, list), 'Prediction result for list of single image must be list'
    print (result)
    
```